java-frontend/src/test/files/se/StreamConsumedCheck.java

```diff
@@ -0,0 +1,51 @@
+import java.util.stream.*;
+import java.util.List;
+
+class A {
+
+  void test() {
+    Stream<Integer> stream = Stream.of(1, 2, 3);
+    stream.count(); // flow@stream {{Pipeline is consumed here.}}
+    stream.findAny(); // Noncompliant [[flows=stream]] {{Refactor this code so that this consumed stream pipeline is not reused.}}
+  }
+
+  void intStream() {
+    IntStream range = IntStream.range(0, 10);
+    range.count(); // flow@is {{Pipeline is consumed here.}}
+    range.average(); // Noncompliant [[flows=is]]
+    range.filter(i -> true); // Noncompliant [[flows=is]]
+  }
+
+  void pipeline() {
+    IntStream range = IntStream.range(0, 10);
+    range  // flow@pipe {{Pipeline is consumed here.}}
```

Comment:
shouldn't this second message be reported two lines below ? This is probably due to methodSelect.